crates/sui-core/src/authority.rs

```diff
@@ -1419,6 +1428,10 @@ impl AuthorityState {
             });
         }
 
+        // Cheap validity checks for a transaction, including input size limits.
+        transaction.check_version_supported(epoch_store.protocol_config())?;
+        transaction.validity_check_no_gas_check(epoch_store.protocol_config())?;
```

Comment:
what about dev_inspect?

Reply:
I took a look at  `dev_inspect_transaction_block()` under `crates/sui-sdk`. It seems transaction input checks are not performed intentionally. Maybe a few validity checks should be done, but probably someone who understands the use case of dev_inspect should make changes here.